### PR TITLE
Reduce wait time in test helpers

### PR DESCRIPTION
**Pull request by dbussink**
Instead of waiting already before we check, immediately check and then start waiting if we need to. This avoids waiting neednessly which slows down tests.

Also shorten the wait loop time to 100ms instead of 1s to speed things up even if we have to wait.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by frouioui — APPROVED** [review]

**Comment by frouioui** [comment]
LGTM! Thank you

**Review by dbussink — COMMENTED** [review]
Nice catch!

File: go/test/endtoend/utils/utils.go [hunk 1]
```diff
@@ -282,6 +282,28 @@ func WaitForKsError(t *testing.T, vtgateProcess cluster.VtgateProcess, ks string
 	}
 }
 
+// WaitForTableDeletions waits for a table to be deleted
+func WaitForTableDeletions(t *testing.T, vtgateProcess cluster.VtgateProcess, ks, tbl string) error {
+	timeout := time.After(60 * time.Second)
+	for {
+		select {
+		case <-timeout:
```
Comment (shlomi-noach) [inline go/test/endtoend/utils/utils.go:290]: This is incorrect: on every single loop iteration you allow a timeout of `60s`. What you really want is a total wait of `60s`. Suggestion: use `context.WithTimeout()`.

Comment (GuptaManan100) [reply go/test/endtoend/utils/utils.go:290]: Okay 👍

**Comment by GuptaManan100** [comment]
I have verified the test isn't flaky anymore with the recent changes. It has passed successively 5 times as compared to failing almost all the time.

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**